task_interface.go

```diff
@@ -20,7 +20,9 @@ type Task interface {
 
 	// Framework tells user task what current epoch is.
 	// This give the task an opportunity to cleanup and regroup.
-	SetEpoch(ctx context.Context, epoch uint64)
+	EnterEpoch(ctx context.Context, epoch uint64)
+
+	ExitEpoch(ctx context.Context, epoch uint64)
```

Comment:
Hmm, both ExitEpoch has empty implementation. And also, is it the context in EnterEpoch can be used to signal the doneness of epoch? I think we should have one instead of both, as they are overlapping.